From 45df4f09e9145dd7bfe7224b0fb40ae97ecc945c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Fri, 5 Jun 2020 17:34:31 +0200
Subject: [PATCH 09/19] winex11.drv: Check for EWMH _NET_ACTIVE_WINDOW support.

CW-Bug-Id: #19553
CW-Bug-Id: #19644
Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=2155

 100.0% dlls/winex11.drv/
diff --git a/dlls/winex11.drv/x11drv.h b/dlls/winex11.drv/x11drv.h
index e7633d16acb..d7ed09f5295 100644
--- a/dlls/winex11.drv/x11drv.h
+++ b/dlls/winex11.drv/x11drv.h
@@ -438,6 +438,7 @@ static inline size_t get_property_size( int format, unsigned long count )
 
 struct x11drv_ewmh_data
 {
+    int has__net_active_window : 1;
     int has__net_wm_moveresize : 1;
 };
 
@@ -499,6 +500,7 @@ enum x11drv_atoms
     XATOM__ICC_PROFILE,
     XATOM__KDE_NET_WM_STATE_SKIP_SWITCHER,
     XATOM__MOTIF_WM_HINTS,
+    XATOM__NET_ACTIVE_WINDOW,
     XATOM__NET_STARTUP_INFO_BEGIN,
     XATOM__NET_STARTUP_INFO,
     XATOM__NET_SUPPORTED,
diff --git a/dlls/winex11.drv/x11drv_main.c b/dlls/winex11.drv/x11drv_main.c
index 8f82ad7efde..c04369be88b 100644
--- a/dlls/winex11.drv/x11drv_main.c
+++ b/dlls/winex11.drv/x11drv_main.c
@@ -155,6 +155,7 @@ static const char * const atom_names[NB_XATOMS - FIRST_XATOM] =
     "_ICC_PROFILE",
     "_KDE_NET_WM_STATE_SKIP_SWITCHER",
     "_MOTIF_WM_HINTS",
+    "_NET_ACTIVE_WINDOW",
     "_NET_STARTUP_INFO_BEGIN",
     "_NET_STARTUP_INFO",
     "_NET_SUPPORTED",
@@ -554,6 +555,8 @@ static void x11drv_ewmh_init(void)
     TRACE( "EWMH _NET_SUPPORTED:\n" );
     for (i = 0; i < supported_count; ++i)
     {
+        if (supported[i] == x11drv_atom(_NET_ACTIVE_WINDOW))
+            ewmh.has__net_active_window = 1;
         if (supported[i] == x11drv_atom(_NET_WM_MOVERESIZE))
             ewmh.has__net_wm_moveresize = 1;
 
-- 
2.43.0

